### PR TITLE
Add pf::api methods view_person,modify_person,add_person

### PR DESCRIPTION
**Pull request by jrouzierinverse**
## Description

Added Crud minus the delete to the webapi
## Impacts

The web api
## NEWS file entries

New Features
++++++++++++
- New api method for adding a person
- New api method for viewing a person
- New api method for modifying a person
## Delete branch after merge

NO

### CODE REVIEW

**Comment by jrouzierinverse** [comment]
Review me

File: lib/pf/api.pm [hunk 1]
```diff
@@ -459,6 +461,127 @@ sub validate_argv {
...
+
+=head2 add_person
+
+Add a new person
+
+=cut
+
+sub add_person : Public {
+    my ($class, %params) = @_;
+    my $logger = pf::log::get_logger();
```
Comment (fdurand) [inline lib/pf/api.pm:641]: I would like to have something like that:
    my @require = qw(pid);
    my @found = grep {exists $params{$_}} @require;
    return unless validate_argv(\@require,  \@found);

File: lib/pf/api.pm [hunk 2]
```diff
@@ -459,6 +461,127 @@ sub validate_argv {
...
+
+=head2 view_person
+
+View a person entry
+
+=cut
+
+sub view_person : Public {
+    my ($class,$pid) = @_;
+    my $logger = pf::log::get_logger();
```
Comment (fdurand) [inline lib/pf/api.pm:702]: I would like to have something like that:
    my @require = qw(pid);
    my @found = grep {exists $params{$_}} @require;
    return unless validate_argv(\@require,  \@found);

File: lib/pf/api.pm [hunk 3]
```diff
@@ -459,6 +461,127 @@ sub validate_argv {
...
+}
+
+=head2 modify_person
+
+Modify an existing person
+
+=cut
+
+sub modify_person : Public {
+    my ($class, %params) = @_;
```
Comment (fdurand) [inline lib/pf/api.pm:724]: I would like to have something like that:
    my @require = qw(pid);
    my @found = grep {exists $params{$_}} @require;
    return unless validate_argv(\@require,  \@found);

**Comment by jrouzierinverse** [comment]
Please review

**Comment by jrouzierinverse** [comment]
Please review

**Pull request merged**